app/server/models/credit_transfer.py

```diff
@@ -0,0 +1,269 @@
+
+    uuid            = db.Column(db.String, unique=True)
+
+    resolved_date   = db.Column(db.DateTime)
+    transfer_amount = db.Column(db.Integer)
+
+    transfer_type   = db.Column(db.Enum(TransferTypeEnum))
+    transfer_status = db.Column(db.Enum(TransferStatusEnum), default=TransferStatusEnum.PENDING)
+    transfer_mode   = db.Column(db.Enum(TransferModeEnum))
+    transfer_use    = db.Column(JSON)
+
+    resolution_message = db.Column(db.String())
+
+    blockchain_task_id = db.Column(db.Integer)
+
+    token_id        = db.Column(db.Integer, db.ForeignKey(Token.id))
+
+    sender_transfer_account_id       = db.Column(db.Integer, db.ForeignKey("transfer_account.id"))
+    recipient_transfer_account_id    = db.Column(db.Integer, db.ForeignKey("transfer_account.id"))
+
+    sender_blockchain_address_id    = db.Column(db.Integer, db.ForeignKey("blockchain_address.id"))
+    recipient_blockchain_address_id = db.Column(db.Integer, db.ForeignKey("blockchain_address.id"))
+
+    sender_user_id    = db.Column(db.Integer, db.ForeignKey("user.id"))
+    recipient_user_id = db.Column(db.Integer, db.ForeignKey("user.id"))
+
+    blockchain_transactions = db.relationship('BlockchainTransaction', backref='credit_transfer', lazy=True)
+
+    attached_images = db.relationship('UploadedImage', backref='credit_transfer', lazy=True)
+
+    @hybrid_property
+    def blockchain_status(self):
+        if self.blockchain_task_id:
+            task = get_blockchain_task(self.blockchain_task_id)
+
+            return task.get('status', 'ERROR')
+        else:
+            return 'UNKNOWN'
+
+        # if len(self.uncompleted_blockchain_tasks) == 0:
+        #     return 'COMPLETE'
+        #
+        # if len(self.pending_blockchain_tasks) > 0:
+        #     return 'PENDING'
+        #
+        # if len(self.failed_blockchain_tasks) > 0:
+        #     return 'ERROR'
+        #
+        # return 'UNKNOWN'
+
+
+    @hybrid_property
+    def blockchain_status_breakdown(self):
+
+        required_task_dict = {x: {'status': 'UNKNOWN', 'hash': None} for x in self._get_required_blockchain_tasks()}
+
+        for transaction in self.blockchain_transactions:
+            status_hierarchy = ['UNKNOWN', 'FAILED', 'PENDING', 'SUCCESS']
+            task_type = transaction.transaction_type
+
+            current_status = required_task_dict.get(task_type).get('status')
+            proposed_new_status = transaction.status
+
+            try:
+                if current_status and status_hierarchy.index(proposed_new_status) > status_hierarchy.index(current_status):
+                    required_task_dict[task_type]['status'] = proposed_new_status
+                    required_task_dict[task_type]['hash'] = transaction.hash
+            except ValueError:
+                pass
+
+        return required_task_dict
+
+    @hybrid_property
+    def pending_blockchain_tasks(self):
+        return self._find_blockchain_tasks_with_status_of('PENDING')
```

Comment:
these should also be an enum

Reply:
so I've shifted away from using an ENUM class directly in the db, because they blow up in a bad way during migrations. My compromise is to use a string hybrid property with checks on the setter (see eth_worker >sql_persistence > models.py, line 178)

`@hybrid_property
    def status(self):
        return self._status

    @status.setter
    def status(self, value):

        if value not in STATUS_STRING_TO_INT:
            raise ValueError('Status {} not allowed. (Must be {}'.format(value, STATUS_STRING_TO_INT))

        self._status = value
`

does that seem reasonable to u? it's a relatively new thing i've been trying out, only in the last month.

Reply:
what causes it to blow up? aren't they just strings?

Reply:
>  aren't they just strings?
I'm pretty sure ENUMS are their own datatype in postgres. To be honest it's been ages since i looked at this (we haven't had to upgrade ENUMS in a while), and my understanding of db migrations wasn't great back then, but from memory this was the issue we were running into:

https://stackoverflow.com/questions/14845203/altering-an-enum-field-using-alembic

Might be worth re-exploring?